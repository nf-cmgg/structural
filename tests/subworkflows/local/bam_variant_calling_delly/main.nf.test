nextflow_workflow {

    name "Test Workflow BAM_VARIANT_CALLING_DELLY"
    script "subworkflows/local/bam_variant_calling_delly/main.nf"
    workflow "BAM_VARIANT_CALLING_DELLY"

    tag "subworkflows"
    tag "subworkflows_local"
    tag "bam_variant_calling_delly"

    config "./nextflow.config"

    test("homo_sapiens") {

        when {
            workflow {
                """
                input[0] = Channel.of([
                    [id:"test", sample:"test", sex:"male"],
                    file(params.cram1, checkIfExists: true),
                    file(params.crai1, checkIfExists: true)
                ])
                input[1] = Channel.value([
                    [id:"fasta"],
                    file(params.fasta, checkIfExists: true)
                ])
                input[2] = Channel.value([
                    [id:"fai"],
                    file(params.fai, checkIfExists: true)
                ])
                input[3] = Channel.fromPath("\${params.svync_configs}/*.yaml", checkIfExists:true).collect()
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(
                    path(workflow.out.delly_vcfs[0][1]).linesGzip[-4..-1],
                    file(workflow.out.delly_vcfs[0][2]).name,
                    workflow.trace.succeeded().size(),
                    workflow.trace.failed().size()
                ).match() }
            )
        }

    }

}
