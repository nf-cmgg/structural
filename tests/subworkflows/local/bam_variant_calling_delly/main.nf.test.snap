{
    "homo_sapiens": {
        "content": [
            [
                "chr16\t86933\tDEL00000000\tT\t<DEL>\t120\tLowQual\tPRECISE;SVTYPE=DEL;SVMETHOD=EMBL.DELLYv1.1.6;END=1349692;PE=0;MAPQ=0;CT=3to5;CIPOS=-9,9;CIEND=-9,9;SRMAPQ=60;INSLEN=0;HOMLEN=9;SR=2;SRQ=0.986667;SVINSSEQ=AAAACAAAAAAAAAAAAAAAAAAAAAAAAAAAATATATATATATATATATATATATATATATATATATATACACATACATATATACGGTTGATTTTTACATATTGATCTTGTATCTTGTAACCTTGCTGAACTTGTTCATTAGTTCTAAT;CE=1.61868\tGT:GL:GQ:FT:RCL:RC:RCR:RDCN:DR:DV:RR:RV\t1/1:-24.3989,-2.10615,0:21:PASS:0:31692:19715:3:0:0:0:7",
                "chr16\t1077371\tINV00000001\tT\t<INV>\t58\tLowQual\tIMPRECISE;SVTYPE=INV;SVMETHOD=EMBL.DELLYv1.1.6;END=1078502;PE=2;MAPQ=29;CT=5to5;CIPOS=-392,392;CIEND=-392,392\tGT:GL:GQ:FT:RCL:RC:RCR:RDCN:DR:DV:RR:RV\t0/0:0,-0.521621,-107.701:6:LowQual:40:102:49:2:19:2:0:0",
                "chr16\t1123476\tINV00000002\tA\t<INV>\t180\tPASS\tPRECISE;SVTYPE=INV;SVMETHOD=EMBL.DELLYv1.1.6;END=1604486;PE=0;MAPQ=0;CT=5to5;CIPOS=-3,3;CIEND=-3,3;SRMAPQ=60;INSLEN=0;HOMLEN=3;SR=3;SRQ=0.98;SVINSSEQ=GAATTGCTTGAACACTGCACCACTGCACTCCAGCCTGGGTGACAGAGGAAGACTCTTTCTCCAAAAAAAAAGAATGTTTTCCTACATATATATATATATATATATATATATACACACACACACACACACACACACACACACACACAGTCT;CE=1.88447\tGT:GL:GQ:FT:RCL:RC:RCR:RDCN:DR:DV:RR:RV\t1/1:-38.9839,-3.59622,0:36:PASS:11456:39951:0:7:0:0:0:12",
                "chr16\t1135261\tINS00000003\tC\t<INS>\t299\tPASS\tPRECISE;SVTYPE=INS;SVMETHOD=EMBL.DELLYv1.1.6;END=1135262;SVLEN=27;PE=0;MAPQ=0;CT=NtoN;CIPOS=-2,2;CIEND=-2,2;SRMAPQ=60;INSLEN=27;HOMLEN=2;SR=5;SRQ=1;SVINSSEQ=AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAACCACTGGAAACAGCCAAGAGATCCTTCAAAAAGTGAATGGATAAACCAACTGTAACTCATTCATACAGTGGAACGTTAATCAGCAATTCTAAAAATGAGCTATCAAGTCACAAAAAGACAAAGAAGAACCTTAACACAAAATAACA;CE=1.67205\tGT:GL:GQ:FT:RCL:RC:RCR:RDCN:DR:DV:RR:RV\t1/1:-90.4974,-7.52311,0:75:PASS:11619:25087:13468:2:0:0:0:25"
            ],
            "test.delly.vcf.gz.tbi"
        ],
        "timestamp": "2023-11-28T14:09:41.242187987"
    }
}